core/grant/service.go

```diff
@@ -673,3 +674,7 @@ func getGrantIDs(grants []domain.Grant) []string {
 func (s *Service) GetGrantsTotalCount(ctx context.Context, filters domain.ListGrantsFilter) (int64, error) {
 	return s.repo.GetGrantsTotalCount(ctx, filters)
 }
+
+func (s *Service) ListUserRoles(ctx context.Context, filters domain.ListGrantsFilter) ([]string, error) {
+	return s.repo.ListUserRoles(ctx, filters)
```

Comment:
let's add validation if `filter.Owner` is empty or not before calling repo.ListUserRoles